system-test/datastore.ts

```diff
@@ -3295,6 +3296,221 @@ async.each(
           assert.strictEqual(entity, undefined);
         });
       });
+      describe('Datastore mode data transforms', () => {
```

Comment:
Does this single test really cover the whole feature? (I dont see anything with `setToServerValue: false` for example. And aren't non-int values supported?)

Reply:
We can add tests to cover `setToServerValue: false` and minimum/maximum etc. for strings. There are tons of permutations/combinations and I don't think we can cover them all, but we can definitely cover these.

maximum/minimum/increment/appendMissingElements/removeAllFromArray proto values can technically accept any type so there really are tons of combinations. In this feature we use encodeValue function to take the user supplied value and encode it into the proto format. The proto supports the generic `value` type so to match we allow users to supply `any` type to support all the different input possibilities.